saleor/warehouse/availability.py

```diff
@@ -35,12 +35,13 @@ def check_stock_quantity(variant: "ProductVariant", country_code: str, quantity:
     If so - returns None. If there is less stock then required raise InsufficientStock
     exception.
     """
-    stocks = Stock.objects.get_variant_stocks_for_country(country_code, variant)
-    if not stocks:
-        raise InsufficientStock(variant)
+    if True: #variant.track_inventory:
+        stocks = Stock.objects.get_variant_stocks_for_country(country_code, variant)
+        if not stocks:
+            raise InsufficientStock(variant)
 
-    if variant.track_inventory and quantity > _get_available_quantity(stocks):
-        raise InsufficientStock(variant)
+        if variant.track_inventory and quantity > _get_available_quantity(stocks):
```

Comment:
If we change the first condition as I suggested then we won't need to check `if variant.track_inventory` again. The second part `if quantity > _get_available_quantity(stocks)` will be enough.